cityofaustin/joplin
# 4099 perm edit view

# Description

Exit criteria: 
As an editor/publisher in a department

> I have permission to view the edit page for any page in my department
> I do not have permission to view the edit page for any page not in my department
> I have permission to save a draft of any page in my department
> I do not have permission to save a draft of any page not in my department

I added Page View Restrictions to the pages (https://www.laceyhenschel.com/blog/2018/8/1/search-history-adding-page-privacy-in-wagtail-programmatically) when pages are created. There is a migration to apply the restrictions to the pages that already exist in production. 

The BeforeEditHook checks to see if the user has permissions (aka is in the group that matches the PageViewPermissions' groups) and if they do not, there is a 404. Because the search results are filtered based on department, this will only happen if someone tries to access a page with a link directly. 








# Types of changes



# How can this be tested?



 

https://joplin-pr-4099-perm-edit-view.herokuapp.com/
editor@austintexas.gov pw:e
member of aph

mod@austintexas.io pw: m
member of arr

a way I've been testing is to navigate to a page the user has permissions for and then change the url to a page above or below to test a different department's page. 

I also double checked that admins can still see all pages


# Checklist:
